torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -91,6 +91,18 @@ std::tuple<XLATensor, XLATensor> GetPromotedXlaTensorsForBinaryOp(
   return std::make_tuple(tensor1, tensor2);
 }
 
+void CheckBinaryOpTypePromotion(const at::Tensor& out, const at::Tensor& self,
+                                const at::Tensor& other) {
+  at::ScalarType resultType = at::result_type(self, other);
+  XLA_CHECK(at::canCast(/*from=*/resultType, /*to*/ out.scalar_type()));
+}
+
+void CheckBinaryOpTypePromotion(const at::Tensor& out, const at::Tensor& self,
+                                const at::Scalar& other) {
+  at::ScalarType resultType = at::result_type(self, other);
+  XLA_CHECK(at::canCast(/*from=*/resultType, /*to*/ out.scalar_type()));
```

Comment:
There is no unified pytorch API that does a `TORCH_CHECK()` to test this?

Reply:
I didn't find any. 
https://github.com/pytorch/pytorch/blob/495bd5818bc094d4b94b309ded614f666c9ca849/aten/src/ATen/native/TensorIterator.cpp#L139
is python's check for this

Reply:
It is kind of hard for us to reuse this validate_dtype function since building an OperandInfo only for this purpose seems to be a waste.